tasks/kernel_matrix_testing/infra.py

```diff
@@ -190,26 +177,89 @@ def ask_for_ssh() -> bool:
+                # Try replacing and adding the .pub suffix
+                possible_pubkeys = [path.with_suffix(".pub"), Path(f"{os.fspath(path)}.pub")]
+                pubkey = next((p for p in possible_pubkeys if p.is_file()), None)
+                privkey = path
+
+            keyname = get_ssh_key_name(pubkey) if pubkey is not None else None
+            if keyname is None:
+                raise Exit(f"Cannot find a key name in {path}")
+            return {'path': os.fspath(privkey), 'name': keyname, 'aws_key_name': keyname}
+
+        # Key hint is not a file, see if it's a key name
+        for pubkey in glob.glob(os.path.expanduser("~/.ssh/*.pub")):
+            privkey = pubkey[:-4]
+            checked_paths.append(privkey)
+            key_name = get_ssh_key_name(Path(pubkey))
+            if key_name == key_hint:
+                return {'path': privkey, 'name': key_hint, 'aws_key_name': key_hint}
+
+        # Check if it's a key name that's there in the agent
+        agent_keys = get_ssh_agent_key_names(ctx)
+        if key_hint in agent_keys:
+            return {'path': None, 'name': key_hint, 'aws_key_name': key_hint}
+
+        raise Exit(
+            f"Could not find file for ssh key {key_hint}. Looked in {possible_paths}, it's not a path, not a file name nor a key name"
+        )
 
-    # Try direct files
-    for path in possible_paths:
-        if os.path.exists(os.path.expanduser(path)):
-            return path
+    cm = ConfigManager()
+    return cm.config.get("ssh")
 
-    # Ok, no file found with that name. However, maybe we can identify the key by the key name
-    # that's present in the corresponding pub files
 
-    for pubkey in glob.glob(os.path.expanduser("~/.ssh/*.pub")):
-        privkey = pubkey[:-4]
-        possible_paths.append(privkey)  # Keep track of paths we've checked
+def ensure_key_in_agent(ctx: Context, key: SSHKey):
+    info(f"[+] Checking that key {key} is in the SSH agent...")
+    res = ctx.run(f"ssh-add -l | grep {key['name']}")
+    if res is None or not res.ok:
+        if key['path'] is None:
+            raise Exit(f"Key {key} not found in the agent and no path provided to add it")
 
-        with open(pubkey) as f:
-            parts = f.read().split()
+        info(f"[+] Key {key} not present in the agent, adding it")
+        res = ctx.run(f"ssh-add {key['path']}")
+        if res is None or not res.ok:
+            raise Exit(f"Could not add key {key} to the SSH agent")
 
-            # Public keys have three "words": key type, public key, name
-            if len(parts) == 3 and parts[2] == ssh_key:
-                return privkey
 
-    raise Exit(f"Could not find file for ssh key {ssh_key}. Looked in {possible_paths}")
+def ensure_key_in_ec2(ctx: Context, key: SSHKey):
+    info(f"[+] Checking that key {key} is in AWS...")
+    res = ctx.run(
+        f"aws-vault exec sso-sandbox-account-admin -- aws ec2 describe-key-pairs --key-names {key['aws_key_name']}"
```

Comment:
Is this the right AWS account? I thought e2e/test-infra-definitions used a separate account from the sandbox.

Reply:
I think so, in the call to `system-probe.start-microvms` [we use aws/sandbox](https://github.com/DataDog/datadog-agent/blob/guillermo.julian/kmt-setup-improvements/tasks/kernel_matrix_testing/stacks.py#L223) as the environment.